core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerData.scala

```diff
@@ -31,82 +38,77 @@ case class ActivationEntry(id: ActivationId, namespaceId: UUID, invokerName: Ins
 
-    /**
-     * Adds an activation entry.
-     *
-     * @param id identifier to deduplicate the entry
-     * @param update block calculating the entry to add.
-     *               Note: This is evaluated iff the entry
-     *               didn't exist before.
-     * @return the entry calculated by the block or iff it did
-     *         exist before the entry from the state
-     */
-    def putActivation(id: ActivationId, update: => ActivationEntry): ActivationEntry = {
-        activationsById.getOrElseUpdate(id, {
-            val entry = update
-            activationByNamespaceId.getOrElseUpdate(entry.namespaceId, new TrieSet[ActivationEntry]).put(entry, {})
-            activationByInvoker.getOrElseUpdate(entry.invokerName, new TrieSet[ActivationEntry]).put(entry, {})
-            entry
-        })
-    }
+  /**
+    * Get the number of activations for each namespace.
+    *
+    * @return a map (namespace -> number of activations in the system)
+    */
+  def activationCountByNamespace: Future[Map[String, BigInt]] = {
+    (sharedStateNamespaces ? GetTheMap()).mapTo[MapWithCounters].map(_.dataMap)
+  }
 
-    /**
-     * Removes the given entry.
-     *
-     * @param entry the entry to remove
-     * @return The deleted entry or None if nothing got deleted
-     */
-    def removeActivation(entry: ActivationEntry): Option[ActivationEntry] = {
-        activationsById.remove(entry.id).map { x =>
-            activationByNamespaceId.getOrElseUpdate(x.namespaceId, new TrieSet[ActivationEntry]).remove(entry)
-            activationByInvoker.getOrElseUpdate(x.invokerName, new TrieSet[ActivationEntry]).remove(entry)
-            x
-        }
-    }
+  /**
+    * Get the number of activations for each invoker.
+    *
+    * @return a map (invoker -> number of activations queued for the invoker)
+    */
+  def activationCountByInvoker: Future[Map[String, BigInt]] = {
+    (sharedStateInvokers ? GetTheMap()).mapTo[MapWithCounters].map(_.dataMap)
+  }
 
-    /**
-     * Removes the activation identified by the given activation id.
-     *
-     * @param aid activation id to remove
-     * @return The deleted entry or None if nothing got deleted
-     */
-    def removeActivation(aid: ActivationId): Option[ActivationEntry] = {
-        activationsById.get(aid).flatMap(removeActivation)
+  /**
+    * Adds an activation entry.
+    *
+    * @param id     identifier to deduplicate the entry
+    * @param update block calculating the entry to add.
+    *               Note: This is evaluated iff the entry
+    *               didn't exist before.
+    * @return the entry calculated by the block or iff it did
+    *         exist before the entry from the state
+    */
+  def putActivation(id: ActivationId, update: => ActivationEntry): ActivationEntry = {
+    activationsById.getOrElseUpdate(id, {
+      val entry = update
+      sharedStateNamespaces ! IncreaseCounter(entry.namespaceId.asString, 1)
```

Comment:
I'm trying to understand the need to track `namespaces` and `invokers` in the LB. Is it related to the tenant isolation ?